frontend/app/src/store/blockchain/tokens.ts

```diff
@@ -39,6 +46,17 @@ export const useBlockchainTokensStore = defineStore('blockchain/tokens', () => {
     );
   };
 
+  const setState = (chain: TokenChains, data: EvmTokensRecord) => {
+    const tokensVal = { ...get(tokensState) };
+    set(tokensState, {
+      ...tokensVal,
+      [chain]: {
+        ...tokensVal[chain],
+        ...data
```

Comment:
Do we really need this? doesn't data have all the detected tokens?

Reply:
so the setState happens per address call, so the data returned there is data on 1 address only